components/select/Select.razor.cs

```diff
@@ -112,6 +115,12 @@ public IEnumerable<TItem> DataSource
                         OnDataSourceChanged?.Invoke();
 
                         _datasource = value;
+                        if (_optionsHasInitialized)
+                        {
+
+                            CreateDeleteSelectOptions();
+                            _alredyRecreatedSelectOptions = true;
+                        }
```

Comment:
`hasChanged = !value.SequenceEqual(_datasource);` is not enough to detect the datasource change. If we always modify the same datasource object(by for example List<T>.Add() and List<T>.Remove()), then `hasChanged` is always `false`.